canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp

```diff
@@ -36,11 +36,19 @@ class SimpleSlave : public canopen::BasicSlave
 {
 public:
   using BasicSlave::BasicSlave;
+  ~SimpleSlave()
+  {
+    if (message_thread.joinable())
+    {
+      message_thread.join();
+    }
+  }
 
 protected:
+  std::thread message_thread;
```

Comment:
```suggestion
  std::thread message_thread = nullptr;
```

`nullptr` is valid in C++, not sure about python. I think it was only `null`;